astropy/timeseries/periodograms/lombscargle/tests/test_lombscargle.py

```diff
@@ -67,8 +68,11 @@ def test_autofrequency(
 def test_all_methods(
     data, method, center_data, fit_mean, errors, with_units, normalization
 ):
-    if method == "scipy" and (fit_mean or errors != "none"):
-        return
+    if method == "scipy":
+        if fit_mean and SCIPY_LT_1_15:
+            pytest.skip("SciPy 1.15+ required for using `fit_mean=True`")
+        elif (errors == "full" or errors == "partial" and normalization == "psd"):
+            pytest.skip("scipy method only supports uniform uncertainties dy")
```

Comment:
This is no longer true in scipy 1.15 – should we plumb through nonuniform uncertainties for scipy 1.15 as well?

Reply:
That would certainly be worth a stab; gonna look into this and see if there is a chance to get it in before feature-freeze as well.

Reply:
Yes, passing `w / w.sum()` as in the other methods as `weights` seems to work, just the `psd` normalisation will need some more investigation.